packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboardNavigation.ts

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react';
+import { gridContainerSizesSelector } from '../../../components/GridViewport';
+import { GRID_CELL_NAVIGATION_KEYDOWN } from '../../../constants/eventsConstants';
+import { GridApiRef } from '../../../models/api/gridApiRef';
+import { GridNavigationApi } from '../../../models/api/gridNavigationApi';
+import { GridCellIndexCoordinates } from '../../../models/gridCell';
+import { GridCellParams } from '../../../models/params/gridCellParams';
+import {
+  isArrowKeys,
+  isEnterKey,
+  isHomeOrEndKeys,
+  isPageKeys,
+  isSpaceKey,
+  isTabKey,
+} from '../../../utils/keyboardUtils';
+import { useGridApiMethod } from '../../root/useGridApiMethod';
+import { optionsSelector } from '../../utils/optionsSelector';
+import { visibleGridColumnsLengthSelector } from '../columns/gridColumnsSelector';
+import { useGridSelector } from '../core/useGridSelector';
+import { useGridState } from '../core/useGridState';
+import { gridPaginationSelector } from '../pagination/gridPaginationSelector';
+import { gridRowCountSelector } from '../rows/gridRowsSelector';
+import { useLogger } from '../../utils/useLogger';
+import { useGridApiEventHandler } from '../../root/useGridApiEventHandler';
+
+const getNextCellIndexes = (code: string, indexes: GridCellIndexCoordinates) => {
+  if (!isArrowKeys(code)) {
+    throw new Error('Material-UI: The first argument (code) should be an arrow key code.');
+  }
+
+  if (code === 'ArrowLeft') {
```

Comment:
It is more of a personal preference but it may look cleaner if instead of `if`s we have a `switch / case` statement.

Reply:
Why did you resolve this? I don't see a comment or a change?

Reply:
yes I can do but I'm trying to focus on the scope of this PR to maintain it small.
I wanted to refactor this with the column header accessibility

Reply:
Haha, I had to restrain myself to not leave comment on parts of the code that was moved around 😁

Reply:
yes otherwise it might stop ppl from trying to cleanup existing code.